packages/react/src/use-chat.ui.test.tsx

```diff
@@ -2314,6 +2314,71 @@ describe('chat instance changes', () => {
 
     expect(screen.queryByTestId('message-0')).not.toBeInTheDocument();
   });
+
+  it('should handle streaming correctly when the id changes', async () => {
```

Comment:
This test is largely copied from the test added in #7387: https://github.com/vercel/ai/blob/68d88037b17d14d55ac0d2a727824fa2449b92af/packages/react/src/use-chat.ui.test.tsx#L2352-L2415